python/nano/tutorial/training/pytorch-lightning/lightning_train_bf16.py

```diff
@@ -0,0 +1,110 @@
+        loss = self.criterion(output, y)
+        pred = torch.argmax(output, dim=1)
+        acc = torch.sum(y == pred).item() / (len(y) * 1.0)
+        metrics = {'test_acc': acc, 'test_loss': loss}
+        self.log_dict(metrics)
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.parameters(), lr=0.01, momentum=0.9, weight_decay=5e-4)
+
+
+def create_dataloaders():
+    train_transform = transforms.Compose([transforms.Resize(256),
+                                          transforms.RandomCrop(224),
+                                          transforms.RandomHorizontalFlip(),
+                                          transforms.ColorJitter(brightness=.5, hue=.3),
+                                          transforms.ToTensor(),
+                                          transforms.Normalize([0.485, 0.456, 0.406],
+                                                               [0.229, 0.224, 0.225])])
+    val_transform = transforms.Compose([transforms.Resize(256),
+                                        transforms.CenterCrop(224),
+                                        transforms.ToTensor(),
+                                        transforms.Normalize([0.485, 0.456, 0.406],
+                                                             [0.229, 0.224, 0.225])])
+
+    # Apply data augmentation to the tarin_dataset
+    train_dataset = OxfordIIITPet(root="/tmp/data", transform=train_transform, download=True)
+    val_dataset = OxfordIIITPet(root="/tmp/data", transform=val_transform)
+
+    # obtain training indices that will be used for validation
+    indices = torch.randperm(len(train_dataset))
+    val_size = len(train_dataset) // 4
+    train_dataset = torch.utils.data.Subset(train_dataset, indices[:-val_size])
+    val_dataset = torch.utils.data.Subset(val_dataset, indices[-val_size:])
+
+    # prepare data loaders
+    train_dataloader = DataLoader(train_dataset, batch_size=32)
+    val_dataloader = DataLoader(val_dataset, batch_size=32)
+
+    return train_dataloader, val_dataloader
+
+
+if __name__ == "__main__":
+
+    model = MyLightningModule()
+    train_loader, val_loader = create_dataloaders()
+
+    # Bfloat16 Training
+    #
+    # BFloat16 is a custom 16-bit floating point format for machine learning
+    # that’s comprised of one sign bit, eight exponent bits, and seven mantissa bits.
+    # BFloat16 has a greater "dynamic range" than FP16. This means it is able to
+    # improve numerical stability than FP16 while delivering increased performance
+    # and reducing memory usage.
+    #
+    # Intel Extension for Pytorch (a.k.a IPEX) support optimizer fusion that can reduce
+    # the number of operators that will be excuted, and reduces overhead time for FP32/BF16
+    #
+    # In BigDL-Nano, you can easily enable both IPEX optimizer fusion
+    # and BFloat16 Mixed precision by setting use_ipex=True and precision='bf16'
+    trainer = Trainer(max_epochs=5, use_ipex=True, precision='bf16')
```

Comment:
does bf16 has to be used with ipex?

Reply:
No, user can just set precision to 16/32/64/'bf16' , which will enable original PyTorch-Lightning precision plugin.

Reply:
ipex + bf16 is better than plain bf16. In this example we show how to use them both (line 106-108 for plain bf16 and line 111-113 for ipex + bf16) should we split it to 2 examples?